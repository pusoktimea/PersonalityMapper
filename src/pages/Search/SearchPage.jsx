import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import SearchForm from 'components/SearchForm';

import './search-page.scss';

class SearchPage extends PureComponent {
  static propTypes = {
    isSearchInProgress: PropTypes.bool.isRequired,
    isSearchComplete: PropTypes.bool.isRequired,
    onDoSearch: PropTypes.func.isRequired,
    searchResults: PropTypes.array.isRequired,
    doResetSearch: PropTypes.func.isRequired,
    location: PropTypes.object.isRequired,
    isSideBarMinimised: PropTypes.bool
  };

  render() {
    const {
      isSearchInProgress,
      isSearchComplete,
      onDoSearch,
      searchResults,
      doResetSearch,
      location,
      isSideBarMinimised
    } = this.props;
    const baseClass = 'main-content';

    return (
      <div className={cx('search-page', baseClass, isSideBarMinimised && `${baseClass}--stretched`)}>
        <h2 className="title">Search</h2>
        <SearchForm
          isSearchInProgress={isSearchInProgress}
          isSearchComplete={isSearchComplete}
          onDoSearch={onDoSearch}
          searchResults={searchResults}
          doResetSearch={doResetSearch}
          location={location}
        />
      </div>
    );
  }
}

export default SearchPage;
