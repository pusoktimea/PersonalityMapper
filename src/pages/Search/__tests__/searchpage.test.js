import 'raf/polyfill';

import React from 'react';
import renderer from 'react-test-renderer';

import SearchPage from '../SearchPage';

const onDoSearch = jest.fn();
const pageProps = {
  onDoSearch: onDoSearch,
  location: {search: ''},
  isSearchInProgress: false,
  isSearchComplete: false,
  searchResults: [],
  doResetSearch: jest.fn()
};

describe('Search Page Test', () => {
  it('initial load, renders into DOM and matches snapshot', () => {
    expect(renderer.create(
      <SearchPage {...pageProps} />).toJSON()
    ).toMatchSnapshot();
  });

  it('with search results, renders into DOM and matches snapshot', () => {
    expect(renderer.create(
      <SearchPage {...{
        ...pageProps,
        isSearchComplete: true,
        searchResults: [{
          count: 1,
          table: 'foo',
          field: 'bar'
        }, {
          count: 100,
          table: 'lorem',
          field: 'ipsum'
        }]
      }} />).toJSON()
    ).toMatchSnapshot();
  });
});
