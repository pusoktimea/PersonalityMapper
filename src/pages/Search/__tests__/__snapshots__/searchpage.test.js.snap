// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Page Test initial load, renders into DOM and matches snapshot 1`] = `
<div
  className="search-page main-content"
>
  <h2
    className="title"
  >
    Search
  </h2>
  <div
    className="activate-panel search-page_form"
  >
    <div
      className="activate-panel_body search-page_form"
    >
      <div
        className="activate-input activate-input--light search-page_form_input"
      >
        <input
          className="activate-input_input"
          disabled={false}
          onChange={[Function]}
          placeholder="Search query"
          type="text"
          value=""
        />
      </div>
      <button
        className="activate-button activate-button--primary activate-button--size-default activate-button--disabled search-page_form_button"
        disabled={true}
        onClick={[Function]}
        to={undefined}
      >
        <span
          className="activate-icon fa fa-search"
          id={undefined}
          onClick={undefined}
        />
        Search
      </button>
    </div>
  </div>
</div>
`;

exports[`Search Page Test with search results, renders into DOM and matches snapshot 1`] = `
<div
  className="search-page main-content"
>
  <h2
    className="title"
  >
    Search
  </h2>
  <div
    className="activate-panel search-page_form"
  >
    <div
      className="activate-panel_body search-page_form"
    >
      <div
        className="activate-input activate-input--light search-page_form_input"
      >
        <input
          className="activate-input_input"
          disabled={false}
          onChange={[Function]}
          placeholder="Search query"
          type="text"
          value=""
        />
      </div>
      <button
        className="activate-button activate-button--primary activate-button--size-default activate-button--disabled search-page_form_button"
        disabled={true}
        onClick={[Function]}
        to={undefined}
      >
        <span
          className="activate-icon fa fa-search"
          id={undefined}
          onClick={undefined}
        />
        Search
      </button>
    </div>
  </div>
  <div
    className="activate-panel search-page_results"
  >
    <div
      className="activate-panel_header search-page_results"
    >
      Search Results
    </div>
    <div
      className="activate-panel_body search-page_results"
    >
      <p>
        <strong>
          1
        </strong>
         
        result was
         found in the 
        <strong>
          foo
        </strong>
         table while looking at the 
        <strong>
          bar
        </strong>
         field.
      </p>
      <p>
        <strong>
          100
        </strong>
         
        results were
         found in the 
        <strong>
          lorem
        </strong>
         table while looking at the 
        <strong>
          ipsum
        </strong>
         field.
      </p>
    </div>
  </div>
</div>
`;
